data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/converter/ExportRecordConverter.java

```diff
@@ -0,0 +1,80 @@
+package org.opensearch.dataprepper.plugins.mongo.converter;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import org.opensearch.dataprepper.buffer.common.BufferAccumulator;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.opensearch.OpenSearchBulkActions;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.mongo.configuration.CollectionConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+
+public class ExportRecordConverter extends RecordConverter {
+    private static final Logger LOG = LoggerFactory.getLogger(ExportRecordConverter.class);
+    private static final Duration VERSION_OVERLAP_TIME_FOR_EXPORT = Duration.ofMinutes(5);
+
+    static final String EXPORT_RECORDS_PROCESSED_COUNT = "exportRecordsProcessed";
+    static final String EXPORT_RECORDS_PROCESSING_ERROR_COUNT = "exportRecordProcessingErrors";
+    static final String BYTES_RECEIVED = "bytesReceived";
+    static final String BYTES_PROCESSED = "bytesProcessed";
+    private final PluginMetrics pluginMetrics;
+    private final long exportStartTime;
+    private final Counter exportRecordSuccessCounter;
+    private final Counter exportRecordErrorCounter;
+    private final DistributionSummary bytesReceivedSummary;
+    private final DistributionSummary bytesProcessedSummary;
+
+    public ExportRecordConverter(final BufferAccumulator<Record<Event>> bufferAccumulator,
+                                 final CollectionConfig collectionConfig,
+                                 final PluginMetrics pluginMetrics,
+                                 final long exportStartTime) {
+        super(bufferAccumulator, collectionConfig);
+        this.pluginMetrics = pluginMetrics;
+        this.exportRecordSuccessCounter = pluginMetrics.counter(EXPORT_RECORDS_PROCESSED_COUNT);
+        this.exportRecordErrorCounter = pluginMetrics.counter(EXPORT_RECORDS_PROCESSING_ERROR_COUNT);
+        this.bytesReceivedSummary = pluginMetrics.summary(BYTES_RECEIVED);
+        this.bytesProcessedSummary = pluginMetrics.summary(BYTES_PROCESSED);
+        this.exportStartTime = exportStartTime;
+    }
+
+    @Override
+    String getEventType() {
+        return "EXPORT";
```

Comment:
The event type is meant to represent the logical type used throughout. This should be the existing `DOCUMENT` type:

https://github.com/opensearch-project/data-prepper/blob/9738730b34d18ba707704fac4e9304fbcee6c468/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/EventType.java#L23

Reply:
Done